docs/docs/learn/concepts/accounts/keys.md

```diff
@@ -1,110 +1,168 @@
+If these keys were not used and a new device would be synched there would be no "direct" information available about notes that a user created for other people.
+
+## Tagging keys
+Called master tagging secret key (`tsk_m`) and master tagging public key (`Tpk_m`).
+Used to compute tags in a [tagging note discovery scheme](../../../protocol-specs/private-message-delivery/private-msg-delivery#note-tagging).
+
+:::note
+Tagging note discovery scheme won't be present in our testnet so we are intentionally not providing you with much info yet.
 :::
 
-### Addresses, partial addresses, and public keys
+## Signing keys
 
-When deploying a contract, the address is deterministically derived from the contract code, the constructor arguments, a salt, and a public key:
+As mentioned above signing keys are not defined in protocol because of [account abstraction](../accounts/main#what-is-account-abstraction) and instead the key scheme is defined by the account contract.
 
-```
-partial_address := hash(salt, contract_code, constructor_hash)
-address := hash(public_key, partial_address)
-```
+Usually, an account contract will validate a signature of the incoming payload against a known signing public key.
 
-This public key corresponds to the privacy master key of the account. In order to manage private state, such as receiving an encrypted note, an account needs to share its partial address and public key, along with its address. This allows anyone to verify that the public key corresponds to the intended address. We call the address, partial address, and public key of a user their **complete address**.
+This is a snippet of our Schnorr Account contract implementation, which uses Schnorr signatures for authentication:
 
-Contracts that are not meant to represent a user who handles private state, usually non-account contracts such as applications, do not need to provide a valid public key, and can instead just use zero to denote that they are not expected to receive private notes.
+#include_code entrypoint /noir-projects/noir-contracts/contracts/schnorr_account_contract/src/main.nr rust
 
-:::info
-A side effect of enshrining and encoding privacy keys into the account address is that these keys cannot be rotated if they are leaked. Read more about this in the [account abstraction section](./main.md#encryption-and-nullifying-keys).
-:::
+Still, different accounts may use different signing schemes, may require multi-factor authentication, or _may not even use signing keys_ and instead rely on other authentication mechanisms. Read [how to write an account contract](../../../developers/contracts/writing_contracts/accounts/write_accounts_contract.md) for a full example of how to manage authentication.
 
-### Encryption keys
+Furthermore, and since signatures are fully abstracted, how the key is stored in the contract is abstracted as well and left to the developer of the account contract.
```

Comment:
What do you think is so weird? That there is a section about storing the signing keys?

I was not really sure about the "storing signing keys" section as it seems like it would be a better fit for a new page on how to build an account contract but it was there before so I decided to keep it there instead of nuking it. Would move it in the future if the account contract section comes into existence.

Or what don't you like about the transition? transition from signing keys to how to store them seems fine to me.

Reply:
It just reads a little fun that it is "below we have this" and the right below that we have "below we have this". But can keep it as is.

Reply:
Oh, I see, yes it's retarded. I will sneak the small fix to my other PR.